lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2696,6 +2688,58 @@ bool ActorIsolationChecker::mayExecuteConcurrentlyWith(
   return false;
 }
 
+bool ActorIsolationChecker::isDistributedThunk(ConcreteDeclRef ref,
+                                               Expr *context,
+                                               bool isInAsyncLetInitializer) {
+  auto *FD = dyn_cast_or_null<AbstractFunctionDecl>(ref.getDecl());
+  if (!(FD && FD->isInstanceMember() && FD->isDistributed()))
+    return false;
+
+  if (!isa<SelfApplyExpr>(context))
+    return false;
+
+  auto actor = getIsolatedActor(cast<SelfApplyExpr>(context)->getBase());
```

Comment:
There's some Stateful Stuff :tm: going on in `getIsolatedActor` with closures, where it is looking at closure captures... but those haven't been computed yet if we're in the constraint solver. I suspect that's going to trip things up here.

Reply:
Opted to only checking whether actor is potentially isolated.